main.py

```diff
@@ -14,27 +14,69 @@ def get_name_tag(resource_type, name):
 
 
 ec2 = boto3.resource('ec2')
+# Create VPC
 vpc = ec2.create_vpc(
     CidrBlock='172.32.0.0/16',
     TagSpecifications=get_name_tag('vpc', 'boto3-vpc')
 )
 
+# Create and attach IGW to attach to VPC
+igw = ec2.create_internet_gateway(
+    TagSpecifications=get_name_tag('internet-gateway', 'boto3-internet-gateway')
+)
+vpc.attach_internet_gateway(InternetGatewayId=igw.id)
+
+
+# create private subnet
 private_subnet = vpc.create_subnet(
     CidrBlock='172.32.0.0/24',
     TagSpecifications=get_name_tag('subnet', 'boto3-private-subnet')
 )
 
+# create 2 public subnets
 public_subnets = []
 for i in range(2):
     public_subnets.append(vpc.create_subnet(
         CidrBlock=f'172.32.{1 + i}.0/24',
         TagSpecifications=get_name_tag('subnet', f'boto3-public-subnet-{i + 1}')
     ))
 
+# create route table and route for private subnet
+private_rt = vpc.create_route_table(
+    TagSpecifications=get_name_tag('route-table', 'boto3-private-rt')
+)
+
+private_rt.create_route(
+    DestinationCidrBlock='0.0.0.0/0',
+    GatewayId=igw.id
+)
+
+private_rt.associate_with_subnet(SubnetId=private_subnet.id)
+
+# create route table and route for public subnet
+public_rt = vpc.create_route_table(
+    TagSpecifications=get_name_tag('route-table', 'boto3-public-rt')
+)
+
+public_rt.create_route(
+    DestinationCidrBlock='0.0.0.0/0',
+    GatewayId=igw.id
+)
+
+for public_subnet in public_subnets:
+    public_rt.associate_with_subnet(SubnetId=public_subnet.id)
+
```

Comment:
We could remove the excessive lines here (PEP-8 E303). I suggest autoformatting on save. [VS Code](https://stackoverflow.com/questions/39494277/how-do-you-format-code-on-save-in-vs-code), [Pycharm](https://stackoverflow.com/a/69999169/7257264)